Mage.Sets/src/mage/cards/l/LoafingGiant.java

```diff
@@ -0,0 +1,76 @@
+package mage.cards.l;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.AttacksOrBlocksTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.PreventCombatDamageBySourceEffect;
+import mage.abilities.effects.common.combat.CantBeBlockedByAllSourceEffect;
+import mage.cards.Card;
+import mage.constants.*;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.players.Player;
+
+/**
+ *
+ * @author noahg
+ */
+public final class LoafingGiant extends CardImpl {
+
+    public LoafingGiant(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{R}");
+        
+        this.subtype.add(SubType.GIANT);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(6);
+
+        // Whenever Loafing Giant attacks or blocks, put the top card of your library into your graveyard. If that card is a land card, prevent all combat damage Loafing Giant would deal this turn.
+        this.addAbility(new AttacksOrBlocksTriggeredAbility(new LoafingGiantEffect(), false));
+    }
+
+    public LoafingGiant(final LoafingGiant card) {
+        super(card);
+    }
+
+    @Override
+    public LoafingGiant copy() {
+        return new LoafingGiant(this);
+    }
+}
+
+class LoafingGiantEffect extends OneShotEffect {
+
+    public LoafingGiantEffect() {
+        super(Outcome.UnboostCreature);
+        this.staticText = "Put the top card of your library into your graveyard. If that card is a land card, prevent all combat damage {this} would deal this turn.";
+    }
+
+    public LoafingGiantEffect(final LoafingGiantEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public LoafingGiantEffect copy() {
+        return new LoafingGiantEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player != null) {
+            Card card = player.getLibrary().getFromTop(game);
+            if (card != null) {
+                player.moveCards(card, Zone.GRAVEYARD, source, game);
```

Comment:
You must check move card result. If it was successful then apply new effect.

Reply:
Ah, okay. In that case Saprazzan Breaker is wrong as well, because that's where I copied this from. I'll fix both.

Reply:
You can fix that card too.

Reply:
I'm not sure what would need to be checked here. I tested with an empty library and no errors were thrown, so that's checked. I also tested with a Rest In Peace out and it still didn't do damage when a land was exiled, as per the Gatherer ruling on Aberrant Researcher (http://gatherer.wizards.com/Pages/Card/Details.aspx?multiverseid=409790). Could you give an example scenario where this would error/not work? Thanks for your feedback as I code my first few cards, by the way!

Reply:
I'm asked mtg judges about that. All fine. Loafing Giant will apply second part of ability anyway even with rest in peace (there aren't "if" sentence). So no need to fix it.